src/genie/libs/parser/iosxr/show_dampening.py

```diff
@@ -0,0 +1,241 @@
+                        {Optional('protocol_capsulation'): str, 
+                        Optional('protocol_penalty'): int,
+                        Optional('protocol_suppression'): str,
+                        Optional('protocol_suppression_timer'): int,
+                        Optional('protocol_underlying_state'): str,
+                      },
+                   },
+                }
+        }
+    }
+    
+class ShowImDampeningIntf(ShowImDampeningIntfSchema):
+    """ Parser for show im dampening interface {interface} """
+    
+    cli_command = ['show im dampening interface {interface}']
+    
+    def cli(self, interface=None, output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            if interface:
+                out = self.device.execute(self.cli_command[1].format(
+                                          interface=interface))
+        else:
+            out = output
+        
+        result_dict = {}
+        result_dict.setdefault('interface', {})
+        
+        # TenGigE 0/1/0/0 (0x01180020)
+        # GigabitEthernet0/2/0/0 (0x080002c0)
+        p1 = re.compile(r'^(?P<interface>[a-zA-Z].+[\/\d]+) \(')
+        
+        # Dampening enabled: Penalty 1625, SUPPRESSED (42 secs remaining)
+        p2 = re.compile(r'Dampening\s(?P<stat>\w+): +Penalty\s(?P<pen>\d+), +SUPPRESSED\s\((?P<sup>\d+)')
+        
+        # Dampening enabled: Penalty 0, not suppressed
+        p3 = re.compile(r'Dampening\s(?P<stat_ns>\w+): +Penalty\s(?P<pen_ns>\d+), +not\ssupp.+')
+        
+        # underlying-state:  Up
+        # Underlying state: Down
+        p4 = re.compile(r'^[a-z-A-Z\s]+state:\s+(?P<und_stat>\S+)')
+        
+        #  half-life: 1        reuse:             1000 
+        p5 = re.compile(r'half-life:\s+(?P<half_life>\d+) +reuse:\s+(?P<reuse>\d+)')
+        
+        # suppress:  1500     max-suppress-time: 4 
+        p6 = re.compile(r'suppress:\s+(?P<suppress>\d+) +max-suppress-time:\s(?P<max_suppress>\d+)')
+        
+        # ipv6           ipv6               1625  YES    42s  remaining        Down
+        p7 = re.compile(r'^(?P<prot>[a-z\d<>].{1,12}) +(?P<prot_cap>[a-z\d<>].{1,12})'
+                         ' +(?P<prot_pen>\d+) +(?P<prot_sup>YES|NO) +(?P<prot_sup_time>\d+).'
+                         ' +remaining\s+(?P<prot_state>\S+)')
+        
+        # Dampening not enabled
+        p8 = re.compile(r'^(?P<stat>Dampening not enabled)')
+        
+        for line in out.splitlines():
+    
+            if line:
+                line = line.strip()
+
+            else:
+                continue
+
+            m = p1.match(line)
+            
+            if m:
+                group = m.groupdict()
+                interface=group['interface']
+                if result_dict['interface'].get(interface)==None:
```

Comment:
Same as other parser, we can improve the code flow

``` python
interface_dict = result_dict.setdefault('interface', {}).setdefault(interface, {})

# Then we can use the interface_dict reference in other matches below as well

Reply:
Changed in the newer version. Thanks.